### PR TITLE
Add OSL object and obj instance names

### PR DESCRIPTION
**Pull request by luisbarrancos**

### CODE REVIEW

File: src/appleseed/renderer/modeling/object/object.h [hunk 1]
```diff
@@ -92,9 +92,18 @@ class APPLESEED_DLLSYMBOL Object
     // Return whether surface shaders should be invoked for fully transparent shading points.
     bool shade_alpha_cutouts() const;
 
+    void* get_oiio_name() const;
+
+    // Destructor.
+    ~Object();
```
Comment (est77) [inline src/appleseed/renderer/modeling/object/object.h]: We could try making the destructor protected.

**Comment by dictoon** [comment]
A few remarks:

- Right now there's a bug if an entity's name is changed with `set_name()`
- We need to change the name, in particular we shouldn't mention oiio as it's confusing
- We should consider moving that to the base `Entity` class (and it already has a private impl)

File: src/appleseed/renderer/modeling/entity/entity.h [hunk 2]
```diff
@@ -102,6 +102,9 @@ class APPLESEED_DLLSYMBOL Entity
     void set_name(const char* name);
     const char* get_name() const;
 
+    // Return the name of the object as a pointer to a ustring for OSL.
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/entity/entity.h]: *object* -> *entity*

Comment (dictoon) [inline src/appleseed/renderer/modeling/entity/entity.h]: *ustring* -> *OIIO::ustring*

**Pull request merged**